confighelper/beaconbroadcaster.py

```
@@ -0,0 +1,129 @@
+        self.logger.info(
+            "Sending beacon %s for program %s.", self.beacon_name, self.program_name)
+        now = datetime.datetime.now().isoformat()
+        random_number = os.urandom(16).encode('hex')
+
+        beacon_filename = '%s---%s' % (now, random_number)
+        final_path = os.path.join(self.beacon_path, beacon_filename)
+
+        try:
+            previous_beacons = os.listdir(self.beacon_path)
+            open(final_path, 'a').close()
+
+            for beacon_file in previous_beacons:
+                os.remove(os.path.join(self.beacon_path, beacon_file))
+
+        except Exception as exception:
+            message = 'Could not create beacon file for beacon %s, program %s. %s:%s' % (
+                self.beacon_name, self.program_name, type(exception).__name__,
+                str(exception))
+            self.logger.critical(message)
+            raise BeaconBroadcasterBroadcastException(message)
+
+    def _set_file_permissions(self, path, mode, uid, gid):
+        """ Sets permissions for a file. Will raise an exception if it fails.
+
+        path: The file to modify
+        mode: The mode to set
+        uid: The owner to set
+        gid: The group to set
+        """
+        try:
+            os.chown(path, uid, gid)
+            # Set permissions to xrw-------
+            os.chmod(path, mode)
+
+        except Exception as exception:
+            self.logger.critical('Could not set permissions for file %s for beacon %s for'
+                                 'program %s. %s: %s.', path, self.beacon_name,
+                                 self.program_name, type(exception).__name__, str(exception))
+            raise BeaconBroadcasterInitException
+
+    def _create_directory(self, path):
+        """ Create a directory while handling errors. Will raise an exception if it fails.
+
+        path: The directory to create
+        """
+        path = os.path.abspath(path)
+        if not os.path.isdir(path):
```

What if it is a file?

---

I suppose it should raise an exception.

---

This has been replaced with your create_directory function from gpgmailer.